src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -249,13 +247,13 @@ case class MatrixRead(
   def execute(hc: HailContext): MatrixValue = {
     val hConf = hc.hadoopConf
 
-    val globals = spec.globalsComponent.readLocal(hc, path)(0)
+    val globals = Annotation.copy(typ.globalType, spec.globalsComponent.readLocal(hc, path)(0)).asInstanceOf[Row]
```

Comment:
Maybe you just want readLocal to return a SafeRow instead of an UnsafeRow?

Reply:
I think it's kind of hard to enforce that guarantee, since readLocal is abstract and SafeRow isn't actually a class. Maybe it should be.